bin/backward-compatibility-test.js

```diff
@@ -0,0 +1,217 @@
+  );
+
+  printStep("Cloning backend...");
+  executeCommand(
+    `git clone --depth 1 -b "${BE_GIT_REF}" https://github.com/metabase/metabase.git "${BE_FOLDER}"`,
+  );
+
+  console.log("Building frontend...");
+  executeCommand("yarn install", FE_FOLDER);
+  executeCommand("yarn build", FE_FOLDER, { MB_EDITION: "ee" });
+
+  printStep("Copying frontend build to backend...");
+  const feResourcesClientPath = path.join(
+    FE_FOLDER,
+    "resources",
+    "frontend_client",
+  );
+  const feResourcesSharedPath = path.join(
+    FE_FOLDER,
+    "resources",
+    "frontend_shared",
+  );
+
+  const beResourcesPath = path.join(BE_FOLDER, "resources");
+  fs.mkdirSync(beResourcesPath, { recursive: true });
+
+  const beTargetClientPath = path.join(beResourcesPath, "frontend_client");
+  const beTargetSharedPath = path.join(beResourcesPath, "frontend_shared");
+
+  fs.cpSync(feResourcesClientPath, beTargetClientPath, {
+    recursive: true,
+    force: true,
+  });
+  fs.cpSync(feResourcesSharedPath, beTargetSharedPath, {
+    recursive: true,
+    force: true,
+  });
+
+  printStep("Building uberjar...");
+  const buildUberjarCommand =
+    "clojure -X:drivers:build:build/all :steps '[:version :translations :drivers :uberjar]'";
+  executeCommand(buildUberjarCommand, BE_FOLDER, { MB_EDITION: "ee" });
+}
+
+function start() {
+  printStep("Starting the uberjar...");
+
+  const dbFile = path.join(os.tmpdir(), `metabase-test-${process.pid}.db`);
+
+  const cypressToken = process.env.CYPRESS_ALL_FEATURES_TOKEN;
+  if (typeof cypressToken === "undefined") {
+    console.error(
+      "Error: CYPRESS_ALL_FEATURES_TOKEN environment variable is not set. This is required for the 'start' command.",
+    );
+    process.exit(1);
+  }
+
+  const javaEnv = {
+    MB_DB_FILE: dbFile,
+    MB_CONFIG_FILE_PATH: "",
+    MB_DANGEROUS_UNSAFE_ENABLE_TESTING_H2_CONNECTIONS_DO_NOT_ENABLE: "true",
+    MB_ENABLE_TEST_ENDPOINTS: "true",
+    MB_PREMIUM_EMBEDDING_TOKEN: cypressToken,
+    MB_JETTY_PORT: "4000",
+  };
+  executeCommand(`java -jar "${JAR_PATH}"`, process.cwd(), javaEnv);
+}
+
+async function checkBackendHealth() {
+  try {
+    const response = await fetch("http://localhost:4000/api/health");
+    if (response.ok) {
+      const data = await response.json();
+      return data.status === "ok";
+    }
+  } catch (error) {
+    return false;
+  }
+  return false;
+}
+
+async function waitForBackend() {
+  printStep("Waiting for backend to be ready...");
+  let attempts = 0;
+  const maxAttempts = 120; // wait at most ~2 minutes
+  while (attempts < maxAttempts) {
+    if (await checkBackendHealth()) {
+      printStep("Backend is ready");
+      return;
+    }
+    attempts++;
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+  }
+  console.error("Backend did not become ready after 2 minutes.");
+  process.exit(1);
+}
+
+async function test() {
+  const CWD_FE = FE_FOLDER;
```

Comment:
I'm not sure how I feel about seeing `CWD_FE` here. I see we pass `FE_FOLDER`  and `BE_FOLDER` directly as the second argument in other `executeCommand` calls. Is there a reason not to pass `FE_FOLDER` directly here?